diff --git a/lisp/frame.el b/lisp/frame.el
index b820d5fcd96..99dfbcd5dea 100644
--- a/lisp/frame.el
+++ b/lisp/frame.el
@@ -1043,6 +1043,7 @@ select-frame-set-input-focus
   (raise-frame frame)
   ;; Ensure, if possible, that FRAME gets input focus.
   (when (display-multi-frame-p frame)
+    (message "AZ: x-focus-frame")
     (x-focus-frame frame))
   ;; Move mouse cursor if necessary.
   (cond
diff --git a/src/pgtkterm.c b/src/pgtkterm.c
index 7839c0dbab5..a7dc7c3c615 100644
--- a/src/pgtkterm.c
+++ b/src/pgtkterm.c
@@ -415,8 +415,10 @@ pgtk_raise_frame (struct frame *f)
      It does not work for child frames on X, and it does not work
      on Wayland too. */
   block_input ();
-  if (FRAME_VISIBLE_P (f))
+  if (FRAME_VISIBLE_P (f)) {
+    printf("AZ: %d\n", gtk_widget_get_has_window (FRAME_WIDGET (f)));
     gdk_window_raise (gtk_widget_get_window (FRAME_WIDGET (f)));
+  }
   unblock_input ();
 }
 
@@ -428,6 +430,7 @@ pgtk_lower_frame (struct frame *f)
   if (FRAME_VISIBLE_P (f))
     {
       block_input ();
+      printf("AZ: %d\n", gtk_widget_get_has_window (FRAME_WIDGET (f)));
       gdk_window_lower (gtk_widget_get_window (FRAME_WIDGET (f)));
       unblock_input ();
     }
